### DISCUSSION TITLE
How can I open the url provided by --player-external-http?

### DISCUSSION THREAD

**Discussion by byOsta** [streamlink/streamlink#4986] [Q&A] [answered]
I have tried open it with VLC and with a browser but nothing happens, where/how can I open it?

**Comment by bastimeyer** [comment, marked as answer]
This is nowhere near enough useful information to help you. Which version of Streamlink are you using and which OS are you using? Post the entire log output when setting the `--loglevel=debug` parameter.

The URLs which get logged by Streamlink point towards a locally spawned HTTP server. If your system prevents Streamlink from hosting an HTTP server, then no clients won't be able to access it.

Regardless of which output method you choose, the stream output is exactly the same, which means the output of [`--player-external-http`](https://streamlink.github.io/cli.html#cmdoption-player-external-http) is exactly the same as when piping the stream output directly into the player's stdin pipe (the CLI's default behavior), or when using the `--player-fifo` or `--player-http` options, or when writing the stream to the filesystem and then reading it afterwards. The only difference here is that Streamlink will wait for the first HTTP client (usually your player) until it actually tries to open the stream.

This means that if you simply tell your player to open one of the URLs logged by Streamlink. How you do that depends on your player. Unless your player is unable to access the HTTP server, this will work just fine.

```
$ streamlink --player-external-http youtube.com/hospitalrecords best
[cli][info] Found matching plugin youtube for URL youtube.com/hospitalrecords
[cli][info] Available streams: 144p (worst), 240p, 360p, 480p, 720p, 1080p (best)
[cli][info] Starting server, access with one of:
[cli][info]  http://127.0.0.1:45769/
[cli][info]  http://172.17.0.1:45769/
[cli][info]  http://192.168.100.1:45769/
[cli][info]  http://192.168.178.99:45769/
...
[cli][info] Got HTTP request from libmpv
[cli][info] Opening stream: 1080p (hls)
...
[cli][info] HTTP connection closed
[cli][info] Stream ended
...
^CInterrupted! Exiting...
[cli][info] Closing currently open stream...
```

```
$ mpv http://127.0.0.1:45769/
 (+) Video --vid=1 (h264 1920x1080 30.000fps)
 (+) Audio --aid=1 (aac 2ch 44100Hz)
AO: [pulse] 44100Hz stereo 2ch float
VO: [gpu] 1920x1080 yuv420p

Exiting... (Quit)
```
